packages/java/src/browser/java-decorator.ts

```diff
@@ -0,0 +1,82 @@
+/********************************************************************************
+ * Copyright (C) 2018 RedHat and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, postConstruct } from 'inversify';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { Tree, TreeNode } from '@theia/core/lib/browser/tree/tree';
+import { DepthFirstTreeIterator } from '@theia/core/lib/browser/tree/tree-iterator';
+import { TreeDecorator, TreeDecoration } from '@theia/core/lib/browser/tree/tree-decorator';
+import { EditorManager } from '@theia/editor/lib/browser';
+import { JAVA_LANGUAGE_ID } from '../common';
+
+@injectable()
+export class JavaDecorator implements TreeDecorator {
+
+    readonly id = 'theia-java-decorator';
+
+    @inject(EditorManager)
+    protected readonly editorManager: EditorManager;
+
+    protected readonly emitter = new Emitter<(tree: Tree) => Map<string, TreeDecoration.Data>>();
+
+    @postConstruct()
+    protected init(): void {
+    }
+
+    async decorations(tree: Tree): Promise<Map<string, TreeDecoration.Data>> {
+        return this.collectDecorators(tree);
+    }
+
+    get onDidChangeDecorations(): Event<(tree: Tree) => Map<string, TreeDecoration.Data>> {
+        return this.emitter.event;
+    }
+
+    protected collectDecorators(tree: Tree): Map<string, TreeDecoration.Data> {
+      const result = new Map();
+        if (tree.root === undefined) {
+            return result;
+        }
+        if (!this.isJavaOutline()) {
+            return result;
+        }
+
+        for (const treeNode of new DepthFirstTreeIterator(tree.root)) {
+            const { name } = treeNode;
+            const colon = name.indexOf(':');
+
+            if (colon !== -1 && name.length > colon + 1) {
```

Comment:
I am a bit puzzled; is `name.length > colon + 1` equivalent to `!name.endsWith(':')`? If so, please simplify. Otherwise, can you please explain to me what use-case you want to handle here?